src/Sulu/Bundle/MediaBundle/DependencyInjection/SuluMediaExtension.php

```diff
@@ -407,32 +410,34 @@ public function load(array $configs, ContainerBuilder $container)
             ]
         );
 
-        $this->configureStorage($config, $container, $loader);
         $this->configureFileValidator($config, $container);
 
         $container->registerForAutoconfiguration(MediaPropertiesProviderInterface::class)
             ->addTag('sulu_media.media_properties_provider');
     }
 
-    private function configureStorage(array $config, ContainerBuilder $container, LoaderInterface $loader)
+    private function configureStorage(array $config, ContainerBuilder $container): void
     {
-        $storage = $container->resolveEnvPlaceholders($config['storage'], true);
-        $container->setParameter('sulu_media.media.storage', $storage);
-
-        foreach ($config['storages'] as $storageKey => $storageConfig) {
-            foreach ($storageConfig as $key => $value) {
-                if ($storageKey === $storage) {
-                    $container->setParameter('sulu_media.media.storage.' . $storageKey . '.' . $key, $value);
-                } else {
-                    // Resolve unused ENV Variables of other Adapter
-                    $container->resolveEnvPlaceholders($value, true);
-                }
-            }
-        }
-
-        $loader->load('services_storage_' . $storage . '.xml');
+        $service = $config['storage']['service'];
+        $flySystemService = $container->get($service, ContainerInterface::NULL_ON_INVALID_REFERENCE);
```

Comment:
its required without flysystem configuration it should not generate the container

Reply:
What do you mean, it should fail if the service isn't there?